tests/test_ilastik/test_applets/base/testSerializer.py

```diff
@@ -535,6 +536,51 @@ def iter_dataset_names(name):
         shutil.rmtree(tmp_dir)
 
 
+@pytest.fixture
+def opLabelArray():
+    raw_data = numpy.zeros((256, 256, 256, 1), dtype=numpy.uint32)
+
+    raw_data[0:15, 0:15, 0:15, 0:1] = numpy.ma.masked
+
+    opLabelArrays = OperatorWrapper(OpCompressedUserLabelArray, graph=Graph())
+    opLabelArrays.Input.resize(1)
+    opLabelArrays.Input[0].meta.has_mask = True
+    opLabelArrays.Input[0].meta.axistags = vigra.AxisTags("zyxc")
+    opLabelArrays.Input[0].setValue(raw_data)
+    opLabelArrays.shape.setValue(raw_data.shape)
+    opLabelArrays.eraser.setValue(255)
+    opLabelArrays.deleteLabel.setValue(-1)
+    opLabelArrays.blockShape.setValue((64, 64, 64, 1))
+    return opLabelArrays
+
+
+def testCompression(tmpdir, opLabelArray):
+    h5_filepath_no_compression = tmpdir / "serial_blockslot_no-compression.h5"
+    h5_filepath_compressed = tmpdir / "serial_blockslot_compressed.h5"
+
+    # Create an operator and a serializer to write the data.
+    slotSerializer_no_compression = SerialBlockSlot(opLabelArray.Output, opLabelArray.Input, opLabelArray.nonzeroBlocks)
+    slotSerializer_compressed = SerialBlockSlot(
+        opLabelArray.Output, opLabelArray.Input, opLabelArray.nonzeroBlocks, compression_level=1
+    )
+    # Give it some data.
+    opLabelArray.Input[0][0:1, 0:1, 0:1, 0:1] = 1 * numpy.ones((1, 1, 1, 1), dtype=numpy.uint8)
+
+    with h5py.File(h5_filepath_no_compression, "w") as f:
+        label_group = f.create_group("label_data")
+        slotSerializer_no_compression.serialize(label_group)
+
+    assert h5_filepath_no_compression.exists()
+
+    with h5py.File(h5_filepath_compressed, "w") as f:
+        label_group = f.create_group("label_data")
+        slotSerializer_compressed.serialize(label_group)
+
+    assert h5_filepath_compressed.exists()
+
+    assert h5_filepath_compressed.stat().size * 10 < h5_filepath_no_compression.stat().size
```

Comment:
I'm feeling a bit uneasy of such tests with approximate pass thresholds...

Reply:
hmmm, Would you have another suggestion? It's not exactly straight forward to calculate the expected file size :/ But maybe a mere `<` then?

Reply:
This is not exactly the same, but what about something like the following?

```python
compressed = True

def all_compressed(_name, obj):
    if isinstance(obj, h5py.Dataset) and obj.compression is None:
        compressed = False
        return False

with h5py.File(h5_filepath_no_compression, "w") as f:
    label_group = f.create_group("label_data")
    slotSerializer_no_compression.serialize(label_group)
    label_group.visititems(all_compressed)
    assert compressed
```

Ditto for the other case.

Reply:
cool thanks, I am inspired already :)

Reply:
opted for a slightly simpler version, but otherwise in line with your suggestion.